doc/connectivity/networking/conn_mgr/implementation.rst

```diff
@@ -0,0 +1,471 @@
+
+Once defined, these implementations can be referenced by name and bound to any unbound iface using :c:macro:`CONN_MGR_BIND_CONN`.
+Take care not to accidentally bind two connectivity implementations to a single iface.
+
+Once the iface is bound, :ref:`connectivity control API <conn_mgr_control_api>` functions can be called on the iface, and they will be translated to the corresponding implementation functions in :c:struct:`conn_mgr_conn_api`.
+
+Binding an iface does not directly modify its :c:struct:`iface struct <net_if>`.
+
+Instead, an instance of :c:struct:`conn_mgr_conn_binding` is created and appended an internal :ref:`iterable section <iterable_sections_api>`.
+
+This binding struct will contain a reference to the bound iface, the connectivity implementation it is bound to, as well as a pointer to a per-iface :ref:`context pointer <conn_mgr_impl_ctx>`.
+
+This iterable section can then be iterated over to find out what (if any) connectivity implementation has been bound to a given iface.
+This search process is used by most of the functions in the :ref:`connectivity control API <conn_mgr_control_api>`.
+As such, these functions should be called sparingly, due to their relatively high search cost.
+
+A single connectivity implementation may be bound to multiple ifaces.
+See :ref:`conn_mgr_impl_guidelines_no_instancing` for more details.
+
+.. _conn_mgr_integration_diagram_detailed:
+
+.. figure:: figures/integration_diagram_detailed.svg
+    :alt: A detailed view of how Connection Manager integrates with Zephyr and the application.
+    :figclass: align-center
+
+    A detailed view of how Connection Manager integrates with Zephyr and the application.
+
+    See :ref:`here <conn_mgr_integration_diagram_simple>` for a simplified version.
+
+.. _conn_mgr_impl_ctx:
+
+Context Pointer
+===============
+
+Since a single connectivity implementation may be shared by several zephyr ifaces, each binding instantiates a context container (of :ref:`configurable type <conn_mgr_impl_declaring>`) unique to that binding.
+Each binding is then instantiated with a reference to that container, which implementations can then use to access per-iface state information.
+
+See also :ref:`conn_mgr_impl_guidelines_binding_access` and :ref:`conn_mgr_impl_guidelines_no_instancing`.
+
+.. _conn_mgr_impl_defining:
+
+Defining an implementation
+==========================
+
+A connectivity implementation may be defined as follows:
+
+.. code-block:: c
+
+   /* Create the API implementation functions */
+   int my_connect_impl(struct conn_mgr_conn_binding *const binding) {
+	/* Your implementation goes here */
+   }
+   int my_disconnect_impl(struct conn_mgr_conn_binding *const binding) {
+	/* Your implementation goes here */
+   }
+   void my_init_impl(struct conn_mgr_conn_binding *const binding) {
+	/* Your implementation goes here */
+   }
```

Comment:
@jukkar Picking up what you mentioned in https://github.com/zephyrproject-rtos/zephyr/pull/62126/commits/1afe35927b6d5fa7cc3ee9c3f0cb7725e3389a6b#r1317146403

Would writing a bit more detail in these comments satisfy what you are requesting?

Reply:
Perhaps. It would be nice to see some info what user could do with the binding, it is not clear now.

Reply:
Okay, I've updated the comments to be a bit more specific. But I want to avoid delving into the specifics of what an implementation is expected to do since that is covered by the other sections of this document, and it would distract from the main point of this section